synapse/app/_base.py

```diff
@@ -273,12 +301,20 @@ def listen_tcp(bind_addresses, port, factory, reactor=reactor, backlog=50):
         except error.CannotListenError as e:
             check_bind_error(e, address, bind_addresses)
 
-    return r
+    # IReactorTCP returns an object implementing IListeningPort from listenTCP,
+    # but we know it will be a Port instance.
+    return r  # type: ignore[return-value]
```

Comment:
Maybe I should have used a `cast` here? This is vaguely unsafe in that the implementation, not the interface, defines this. Realistically this is deep in Twisted and I can't imagine this changing.

Reply:
Given that there are not other lines that follow, I think a cast and a type-ignore are broadly equivalent.